### PR TITLE
Continuation of refactor to TypeScript

### PR DESCRIPTION
**Pull request by Misterblue**
This is building on "feature/refactor-to-typescript" branch to merge all of the code from 'master' into a TypeScript based version and doing some merging of build/checking code from vircadia-web-sdk.

The steps are planned to be:

- [x] add ESLint rules from -web-sdk and fix up code
- [x] add style and menu setup from -web master
- [x] add Vuex modules for state
- [x] port .vue files from master

This builds and runs as it did previously with all script code changes to TypeScript.

I did some code re-organization to split the operational pieces into modules for Vuex (which provide the reactable variables for the interface) and modules that include the code and functions for the feature. Most noticable is Renderer which is now a store/renderer/index.ts (which makes some variables available for the Vue components) and /modules/render/(index.ts|renderer.ts|vscene.ts) which provide the actual functionality. This moves all the entity generation code out of the Vuex modules and makes sure the Vuex properties are only updated by mutation routines.

### CODE REVIEW

**Review by digisomni — CHANGES_REQUESTED** [review]

File: src/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * An object that contains string keys to string values
+ * @typedef {Object.<string,string>} KeyValue
+ */
+type KeyValue = { [ key: string ] : string };
+
+/**
+ * A value that can have any value
+ * @typedef ({KeyedCollection|string|number|unknown}) AnyValue
+ */
+type AnyValue = KeyedCollection | string | number | unknown;
```
[the remark this answers is not in the document]
Comment (Misterblue) [reply src/global.d.ts:29]: I don't see the change here. Is it the blank line after?

File: src/components/overlays/OverlayManager.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    "name": overlay,
+                    "overlayStatus": "restored"
+                });
+            }
+        }
+    }
+});
+</script>
+
+function defineComponent(
```
[the remark this answers is not in the document]
Comment (Misterblue) [reply src/components/overlays/OverlayManager.vue:87]: Embarrassment at not deleting old code.

**Comment by Misterblue** [comment]
I incorporated the suggested changes into a commit that I added to this PR. (Retrospectively, I notice I could have combined your changes into a batch commit. I'll figure out how to effectively use Github someday).

File: src/global.d.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (digisomni) [reply src/global.d.ts:29]: Yup, just a new line at the end of file.

**Pull request merged**